src/overlay005/ov5_021F77A8.c

```diff
@@ -45,512 +47,513 @@ typedef struct {
-	{ 0x1A, 0xE8, 0x0, 0x0, 0x1D },
-	{ 0x1A, 0xE8, 0x0, 0x0, 0x1D },
-	{ 0x2, 0x20, 0x3, 0xC0, 0x30 },
-	{ 0x66, 0x21, 0x3, 0xC0, 0x30 },
-	{ 0x8, 0x68, 0x0, 0x0, 0x39 },
-	{ 0x82, 0xC8, 0x0, 0x4, 0x3A },
-	{ 0x6, 0x20, 0x0, 0x40, 0x1C },
-	{ 0x16, 0x20, 0x0, 0x40, 0x1C },
-	{ 0x18, 0x20, 0x61, 0x40, 0x35 },
-	{ 0x6, 0x20, 0x0, 0x0, 0x18 },
-	{ 0x6, 0x20, 0x0, 0x0, 0x18 },
-	{ 0x10, 0x20, 0x28, 0x0, 0x1 },
-	{ 0x10, 0x22, 0x68, 0x0, 0x1 },
-	{ 0x10, 0x20, 0x1, 0x1, 0x28 },
-	{ 0x2, 0x20, 0x21, 0x0, 0x21 },
-	{ 0x1A, 0x20, 0x43, 0x40, 0x20 },
-	{ 0x2, 0xA8, 0x0, 0x4, 0x32 },
-	{ 0x8, 0x68, 0x0, 0x0, 0x29 },
-	{ 0x6, 0x22, 0x4, 0x5, 0x10 },
-	{ 0x6, 0x22, 0x4, 0x5, 0x10 },
-	{ 0x6, 0x26, 0x4, 0x5, 0x10 },
-	{ 0xFA, 0x21, 0x40, 0x4A, 0x20 },
-	{ 0x26, 0x21, 0xC1, 0x30, 0x10 },
-	{ 0x26, 0x21, 0xC1, 0x30, 0x10 },
-	{ 0x82, 0x20, 0x0, 0x1, 0x0 },
-	{ 0x82, 0x22, 0x0, 0x1, 0x0 },
-	{ 0x6, 0x24, 0x0, 0x0, 0x4 },
-	{ 0x6, 0x24, 0x0, 0x0, 0x4 },
-	{ 0x2E, 0x61, 0x81, 0x82, 0xC },
-	{ 0x2E, 0x61, 0x81, 0x82, 0xC },
-	{ 0x6, 0x30, 0x10, 0x8, 0x4 },
-	{ 0x9, 0xAC, 0x0, 0x84, 0x10 },
-	{ 0x9, 0xAC, 0x0, 0x84, 0x10 },
-	{ 0xB, 0xA0, 0x1, 0x80, 0x10 },
-	{ 0xA, 0x21, 0x10, 0x8, 0x0 },
-	{ 0xA, 0x21, 0x14, 0x8, 0x0 },
-	{ 0xE, 0x61, 0x0, 0x0, 0x14 },
-	{ 0x10, 0x22, 0x20, 0x30, 0x10 },
-	{ 0x7A, 0x25, 0x40, 0x0, 0x4 },
-	{ 0xFE, 0x67, 0xE, 0x1, 0x20 },
-	{ 0x2, 0x20, 0x18, 0x8, 0x4 },
-	{ 0x62, 0x21, 0x21, 0x20, 0x10 },
-	{ 0x62, 0x20, 0x1, 0x0, 0x2 },
-	{ 0x12, 0xA8, 0x6A, 0x44, 0x13 },
-	{ 0x2, 0xA8, 0x28, 0x0, 0x30 },
-	{ 0x6, 0x20, 0x11, 0x48, 0x14 },
-	{ 0xA, 0x24, 0x21, 0x40, 0x10 },
-	{ 0x6, 0x24, 0x0, 0x1, 0x14 },
-	{ 0x8E, 0x22, 0xA, 0x1, 0x4 },
-	{ 0x8, 0x20, 0x60, 0x40, 0x31 },
-	{ 0x66, 0x21, 0xE1, 0x0, 0x15 },
-	{ 0x72, 0x23, 0x8, 0x31, 0x0 },
-	{ 0x6A, 0x69, 0x0, 0x0, 0x9 },
-	{ 0x1A, 0x69, 0x20, 0x0, 0x9 },
-	{ 0x2, 0x68, 0x20, 0x0, 0x39 },
-	{ 0x62, 0x21, 0x61, 0x0, 0x15 },
-	{ 0x62, 0x21, 0x61, 0x0, 0x15 },
-	{ 0x62, 0x21, 0x61, 0x40, 0x35 },
-	{ 0x6, 0x22, 0xC, 0x35, 0x10 },
-	{ 0x7, 0x24, 0xC, 0x5, 0x10 },
-	{ 0x2, 0x22, 0x8, 0x11, 0x22 },
-	{ 0xEA, 0x23, 0x48, 0x1, 0x0 },
-	{ 0xE, 0xEE, 0xC, 0x5, 0x10 },
-	{ 0xE, 0x20, 0x61, 0x0, 0x11 },
-	{ 0xB, 0x20, 0x29, 0xC0, 0x34 },
-	{ 0xB, 0x20, 0x29, 0xC0, 0x34 },
-	{ 0xA, 0x68, 0x0, 0x40, 0x1D },
-	{ 0x2, 0x20, 0x52, 0x49, 0x10 },
-	{ 0xF, 0x2E, 0x6C, 0x55, 0x13 },
-	{ 0x82, 0x20, 0x60, 0x0, 0x1 },
-	{ 0x2, 0x20, 0x60, 0x10, 0x5 },
-	{ 0x2, 0x20, 0x60, 0x0, 0x15 },
-	{ 0x12, 0x20, 0x22, 0x1, 0x28 },
-	{ 0x0, 0x22, 0x22, 0x32, 0x28 },
-	{ 0xE, 0xEE, 0xC, 0x5, 0x10 },
-	{ 0x2, 0xA8, 0x50, 0x48, 0x10 },
-	{ 0x2, 0x68, 0x20, 0x0, 0x39 },
-	{ 0x2, 0x68, 0x20, 0x0, 0x39 },
-	{ 0x2, 0x68, 0x20, 0x0, 0x39 },
-	{ 0x2, 0x68, 0x20, 0x0, 0x39 },
-	{ 0x2, 0x68, 0x20, 0x0, 0x39 }
+// I've got next to zero clue what this horseshit does. All I know is that there's one for every species + form, so it's important when increasing the pokedex size
+const UnkStruct_ov5_02200EAC Unk_ov5_02200EAC[FORMS_MAX] = {
```

Comment:
Couple things:

1. This comment doesn't really add anything to the code; get rid of it.
2. We discerned, if I recall, that this is related to learnsets, possibly to the move relearner. It could use with a label to reflect that.
3. Get rid of the comments to the right of these table entries; if you intend to index them according to a value, do so with bracket notation, like so:

```c
const UnkStruct_ov5_02200EAC Unk_ov5_02200EAC[FORMS_MAX] = {
    [0] = { 0x6, 0x20, 0x10, 0x8, 0x4 },
    [1] = { 0x6, 0x20, 0x10, 0x8, 0x4 },
    // etc.
};
```

It looks to me like these are dex numbers, just without the leading `SPECIES_NONE` entry. So, the indices should probably be `SPECIES_` constants, offset by 1.

Reply:
Removed the comments, I decided against using indexing here because I feel like this array should eventually be automatically generated from something like `data.json` for each species

Reply:
I like the idea of building this table from a data file. Good shout. Can you file an issue for it?

Reply:
Done: #146